src/Device/TCodeDevice.cs

```diff
@@ -39,17 +78,20 @@ public TCodeDevice()
 
             _lastNoCollisionTime = Time.time;
             _stringBuilder = new StringBuilder();
+            _deviceReportBuilder = new StringBuilder();
         }
 
-        private string AppendIfChanged(StringBuilder stringBuilder, string axisName, float cmd, ref float lastCmd)
+        private string FormatCommand(string axisName, float cmd)
```

Comment:
Since this is only used with StringBuilders I think it will save some allocations if it uses StringBuilder directly like `AppendIfChanged`

Reply:
Isn't FormatCommand just one string interpolation with the result being returned as a reference?
What I really would like to get rid of is the redundant code in GetDeviceReport.
I might not 100% understand what you mean here.

Reply:
Ah, I get it. I don't think string interpolation causes more allocations than a StringBuilder but it's been a while since I dug into the internal workings of interpolation.

Reply:
I know newer versions of .net heavily optimize interpolation but im pretty sure unity/vam will use `string.Concat` which will create its own StringBuilder.

Reply:
You're right, there's less magic than I thought.

> In C# 9 and earlier, if an interpolated string has the type string, it's typically transformed into a [String.Format](https://learn.microsoft.com/en-us/dotnet/api/system.string.format) method call. The compiler may replace [String.Format](https://learn.microsoft.com/en-us/dotnet/api/system.string.format) with [String.Concat](https://learn.microsoft.com/en-us/dotnet/api/system.string.concat) if the analyzed behavior would be equivalent to concatenation.